Index: component/webui/src/main/resources/locale/social/Webui_en.properties
===================================================================
--- component/webui/src/main/resources/locale/social/Webui_en.properties	(revision 80206)
+++ component/webui/src/main/resources/locale/social/Webui_en.properties	(working copy)
@@ -117,6 +117,7 @@
 UIComposer.input.What_Are_You_Working_On=What are you working on?
 UIComposer.label.Add=Add
 UIComposer.label.Share=Share
+UIComposer.label.ShareStatus=Share My Status
 UIComposer.msg.error.Empty_Message=The message is empty.
 
 #####################################################################################
@@ -349,9 +350,10 @@
 #                             UIUserActivitiesDisplay                               #
 #####################################################################################
 UIUserActivitiesDisplay.label.All_Updates=All Updates
-UIUserActivitiesDisplay.label.Network_Updates=Network Updates
-UIUserActivitiesDisplay.label.Space_Updates=Space Updates
+UIUserActivitiesDisplay.label.Network_Updates=My Connections
+UIUserActivitiesDisplay.label.Space_Updates=My Spaces
 UIUserActivitiesDisplay.label.My_Status=My Status
+UIUserActivitiesDisplay.label.My_Activity_Stream=My Activity Stream
 
 #####################################################################################
 #                              UIAddApplicationSpace                                #
Index: component/webui/src/main/resources/locale/social/Webui_fr.properties
===================================================================
--- component/webui/src/main/resources/locale/social/Webui_fr.properties	(revision 80206)
+++ component/webui/src/main/resources/locale/social/Webui_fr.properties	(working copy)
@@ -118,6 +118,7 @@
 UIComposer.label.Add=Ajouter
 UIComposer.label.Attach=Attacher
 UIComposer.label.Share=Partager
+UIComposer.label.ShareStatus=Partager mon statut
 UIComposer.msg.error.Empty_Message=Le message est vide.
 
 #####################################################################################
@@ -349,9 +350,10 @@
 #                             UIUserActivitiesDisplay                               #
 #####################################################################################
 UIUserActivitiesDisplay.label.All_Updates=Fil d'actualité
-UIUserActivitiesDisplay.label.Network_Updates=Mes contacts
+UIUserActivitiesDisplay.label.Network_Updates=Mes Connexions
 UIUserActivitiesDisplay.label.Space_Updates=Mes espaces
 UIUserActivitiesDisplay.label.My_Status=Mon statut
+UIUserActivitiesDisplay.label.My_Activity_Stream=Mon Flux d'Activité
 
 #####################################################################################
 #                              UIAddApplicationSpace                                #
Index: component/webui/src/main/resources/groovy/social/webui/profile/UIUserActivitiesDisplay.gtmpl
===================================================================
--- component/webui/src/main/resources/groovy/social/webui/profile/UIUserActivitiesDisplay.gtmpl	(revision 80206)
+++ component/webui/src/main/resources/groovy/social/webui/profile/UIUserActivitiesDisplay.gtmpl	(working copy)
@@ -27,6 +27,7 @@
   def labelNetworkUpdates = _ctx.appRes("${uicomponent.id}.label.Network_Updates");
   def labelSpaceUpdates = _ctx.appRes("${uicomponent.id}.label.Space_Updates");
   def labelMyStatus = _ctx.appRes("${uicomponent.id}.label.My_Status");
+  def labelMyActivityStream = _ctx.appRes("${uicomponent.id}.label.My_Activity_Stream");
 
   def selectedDisplayMode = uicomponent.getSelectedDisplayMode();
   def allUpdatesDisplayModeStyle = "LightBlueStyle";
@@ -45,6 +46,7 @@
 
 %>
 <div id="$uicomponent.id" class="$uicomponent.id">
+  <h4 class="ActivityStreamTitle">$labelMyActivityStream</h4>
 <% if (uicomponent.isActivityStreamOwner()) { %>
   <div class="UIAction">
     <a href="javascript:void(0);" onclick="<%= uicomponent.event("ChangeDisplayMode", "ALL_UPDATES");%>" class="ActionButton $allUpdatesDisplayModeStyle">$labelAllUpdates</a>
Index: component/webui/src/main/resources/groovy/social/webui/composer/UIComposer.gtmpl
===================================================================
--- component/webui/src/main/resources/groovy/social/webui/composer/UIComposer.gtmpl	(revision 80206)
+++ component/webui/src/main/resources/groovy/social/webui/composer/UIComposer.gtmpl	(working copy)
@@ -24,6 +24,7 @@
   import org.exoplatform.webui.core.UIPopupWindow;
   import org.exoplatform.webui.form.UIFormTextAreaInput;
   import org.exoplatform.social.webui.composer.UIComposer;
+  import org.exoplatform.social.webui.composer.UIComposer.PostContext;
   import org.exoplatform.social.webui.composer.UIActivityComposerContainer;
   import org.exoplatform.social.webui.composer.UIDefaultActivityComposer;
   import org.exoplatform.social.core.service.LinkProvider;
@@ -38,6 +39,7 @@
   def addLabel = _ctx.appRes("${uicomponent.id}.label.Add");
   def shareLabel = _ctx.appRes("${uicomponent.id}.label.Share");
   def blastInput = "BlastInputSpace";
+  def shareStatusLabel = _ctx.appRes("${uicomponent.id}.label.ShareStatus");
 
   if (uicomponent.getPostContext().toString().equals("USER") && uicomponent.isActivityStreamOwner()) {
     defaultInput = _ctx.appRes("${uicomponent.id}.input.What_Are_You_Working_On");
@@ -54,6 +56,7 @@
       blurColor: null,
       minHeight: null,
       maxHeight: null,
+      padding: null,
       focusHeight: null,
       focusCallback: null,
       blurCallback: null,
@@ -97,6 +100,9 @@
 </script>
   <% uiform.begin() %>
   <div class="BlastContainer">
+    <% if (uicomponent.getPostContext() == PostContext.USER) { %>
+         <h4>$shareStatusLabel</h4>
+    <% } %>
     <div class="$blastInput">
       <% uicomponent.renderChild(UIFormTextAreaInput.class) %>
     </div>
Index: webapp/portlet/src/main/webapp/skin/social/portlet/UIUserActivityStreamPortlet/DefaultStylesheet.css
===================================================================
--- webapp/portlet/src/main/webapp/skin/social/portlet/UIUserActivityStreamPortlet/DefaultStylesheet.css	(revision 80206)
+++ webapp/portlet/src/main/webapp/skin/social/portlet/UIUserActivityStreamPortlet/DefaultStylesheet.css	(working copy)
@@ -39,11 +39,19 @@
 	background: #EBEBEB;
 }
 
+.UIUserActivityStreamPortlet .UIUserActivitiesDisplay .ActivityStreamTitle {
+    background: none repeat scroll 0 0 #F8F8F8;
+    border: 1px solid #E1E1E1;
+    border-radius: 3px 3px 3px 3px;
+    padding: 10px;
+    font-weight: normal;
+}
+
 .UIUserActivityStreamPortlet .UIAction {
 	border: none;
 	border-bottom: 1px solid #e1e1e1;
 	text-align: left;
-	padding: 20px 0px;
+	padding: 0px 0px 20px;
 }
 
 .UIUserActivityStreamPortlet .UIActivitiesLoadButton {
@@ -77,4 +85,4 @@
   list-style: disc inside none;
   padding: 0px 0px 0px 20px;
   margin: 0px 0px 0px 0px;
-}
\ No newline at end of file
+}
Index: webapp/resources/src/main/webapp/skin/social/webui/UIComposer/DefaultSkin.css
===================================================================
--- webapp/resources/src/main/webapp/skin/social/webui/UIComposer/DefaultSkin.css	(revision 80206)
+++ webapp/resources/src/main/webapp/skin/social/webui/UIComposer/DefaultSkin.css	(working copy)
@@ -9,6 +9,11 @@
 	border-radius: 3px;
 }
 
+.UIComposer .BlastContainer h4 {
+    margin: 0 2px 8px;
+    font-weight: normal;
+}
+
 .UIComposer .BlastContainer .BlastInputSpace {
 	margin-bottom: 4px;
 }
@@ -16,9 +21,11 @@
 .UIComposer .BlastContainer textarea {
 	width: 99% !important;
 	border: 1px solid #e1e1e1;
-	font-size: 11px;
+	font-size: 14px;
 	border-radius: 2px;
-	padding: 5px 5px 0px 5px;
+	margin: 0 0 7px;
+	line-height: 20px;
+	padding: 0 0 0 8px;
 }
 
 .UIComposer .BlastContainer .BlastShare {
@@ -57,8 +64,9 @@
 }
 
 .UIComposer .BlastContainer .BlastShare .AttachMents {
-	padding-top: 8px;
+	padding-top: 5px;
 	font-weight: bold;
+	float: left;
 }
 
 .UIComposer .BlastContainer .BlastShare .ShareButton {
@@ -108,4 +116,4 @@
 	line-height: 8px;
 	margin-right: 0;
 	padding: 0 4px 4px;
-}
\ No newline at end of file
+}
Index: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIComposer.js
===================================================================
--- webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIComposer.js	(revision 80206)
+++ webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIComposer.js	(working copy)
@@ -71,6 +71,7 @@
     this.minHeight = params.minHeight || '20px';
     this.focusHeight = params.focusHeight || '35px';
     this.maxHeight = params.maxHeight || '50px';
+    this.padding = params.padding || '11px 0 11px 8px';
     this.focusCallback = params.focusCallback;
     this.blurCallback = params.blurCallback;
     this.keypressCallback = params.keypressCallback;
@@ -89,6 +90,7 @@
     this.composer.value = this.defaultInput;
     this.composer.style.height = this.minHeight;
     this.composer.style.color = this.blurColor;
+    this.composer.style.padding = this.padding;
     this.shareButton.className = 'ShareButtonDisable';
     this.shareButton.disabled = true;
     this.currentValue = this.composer.value;
@@ -105,6 +107,7 @@
         uiComposer.focusCallback();
       }
       uiComposer.composer.style.height = uiComposer.maxHeight;
+      uiComposer.composer.style.padding = null;
       uiComposer.composer.style.color = uiComposer.focusColor;     
     }, false);
 
@@ -112,6 +115,7 @@
       if (uiComposer.composer.value === '') {
         uiComposer.composer.value = uiComposer.defaultInput;
         uiComposer.composer.style.height = uiComposer.minHeight;
+        uiComposer.composer.style.padding = uiComposer.padding;
         uiComposer.composer.style.color = uiComposer.blurColor;
 
         //if current composer is default composer then disable share button
@@ -122,6 +126,7 @@
 
       } else {
         	uiComposer.currentValue = uiComposer.composer.value;
+        	uiComposer.composer.style.padding = uiComposer.padding;
       }
       
       if (uiComposer.blurCallback) {
