Index: component/core/src/main/java/org/exoplatform/social/core/activity/model/Util.java
===================================================================
--- component/core/src/main/java/org/exoplatform/social/core/activity/model/Util.java	(revision 69138)
+++ component/core/src/main/java/org/exoplatform/social/core/activity/model/Util.java	(working copy)
@@ -37,7 +37,7 @@
         if (a1 == null || a2 == null) {
           throw new IllegalArgumentException("Cannot compare null Activity");
         }
-        return (int) (a2.getPostedTime() - a1.getPostedTime());
+        return a2.getPostedTime().compareTo(a1.getPostedTime());
       }
     };
   }
