Index: component/core/src/main/java/org/exoplatform/social/core/space/impl/SocialGroupEventListenerImpl.java
===================================================================
--- component/core/src/main/java/org/exoplatform/social/core/space/impl/SocialGroupEventListenerImpl.java	(revision 76486)
+++ component/core/src/main/java/org/exoplatform/social/core/space/impl/SocialGroupEventListenerImpl.java	(working copy)
@@ -20,9 +20,6 @@
 
 import org.exoplatform.container.ExoContainer;
 import org.exoplatform.container.ExoContainerContext;
-import org.exoplatform.portal.webui.util.Util;
-import org.exoplatform.portal.webui.workspace.UIPortalApplication;
-import org.exoplatform.portal.webui.workspace.UIWorkingWorkspace;
 import org.exoplatform.services.organization.Group;
 import org.exoplatform.services.organization.GroupEventListener;
 import org.exoplatform.social.core.space.model.Space;
@@ -30,8 +27,6 @@
 
 public class SocialGroupEventListenerImpl extends GroupEventListener {
 
-  private SpaceService spaceService;
-
   public SocialGroupEventListenerImpl() {
     // TODO Auto-generated constructor stub
   }
@@ -71,7 +66,9 @@
    *           not remove the group record from the database.
    */
   public void preDelete(Group group) throws Exception {
-    SpaceService spaceSrv = getSpaceService();
+    SpaceService  spaceSrv = (SpaceService) ExoContainerContext.getCurrentContainer()
+      .getComponentInstance(SpaceService.class);
+    
     List<Space> allSpaces = spaceSrv.getAllSpaces();
     String groupId = group.getId();
     String groupIdOfSpace = null;
@@ -81,11 +78,6 @@
         spaceSrv.deleteSpace(space);
       }
     }
-
-    UIPortalApplication uiPortalApp = Util.getUIPortalApplication();
-    UIWorkingWorkspace uiWorkSpace = uiPortalApp.getChild(UIWorkingWorkspace.class);
-    uiWorkSpace.updatePortletsByName("SpacesToolbarPortlet");
-    uiWorkSpace.updatePortletsByName("SocialUserToolBarGroupPortlet");
   }
 
   /**
@@ -100,12 +92,4 @@
    */
   public void postDelete(Group group) throws Exception {
   }
-
-  private SpaceService getSpaceService() {
-    ExoContainer container = ExoContainerContext.getCurrentContainer();
-    if (spaceService == null)
-      spaceService = (SpaceService) container.getComponentInstance(SpaceService.class);
-
-    return spaceService;
-  }
 }
