Index: component/webui/src/main/resources/groovy/social/webui/UIApplicationCategorySelector.gtmpl
===================================================================
--- component/webui/src/main/resources/groovy/social/webui/UIApplicationCategorySelector.gtmpl	(revision 80206)
+++ component/webui/src/main/resources/groovy/social/webui/UIApplicationCategorySelector.gtmpl	(working copy)
@@ -19,12 +19,36 @@
 
 -->
 */
+import org.exoplatform.social.core.service.LinkProvider;
 
 def uicomponentId = uicomponent.id;
 def categoriesLabel = _ctx.appRes("${uicomponentId}.label.categories");
 def categoryList = uicomponent.getApplicationCategoryList();
 
+def rcontext = _ctx.getRequestContext();
+def jsmanager = rcontext.getJavascriptManager();
+jsmanager.importJavascript("eXo.social.Util", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+jsmanager.importJavascript("eXo.social.PortalHttpRequest", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+jsmanager.importJavascript("eXo.social.webui.UIForm", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+jsmanager.importJavascript("eXo.social.webui.UIApplicationCategorySelector", LinkProvider.JAVASCRIPT_RESOURCE_URL);
+
+def applicationCategoryIds = [];
+for (applicationCategory in categoryList) {
+  applicationCategoryIds.add(applicationCategory.displayName);
+}
+def params = """
+  {
+    applicationCategoryIds: '$applicationCategoryIds',
+    allApplicationCategorySize: ${applicationCategoryIds.size()},
+  }
+"""
+jsmanager.addOnLoadJavascript("initUIApplicationCategorySelector");
 %>
+<script type="text/javascript">
+  function initUIApplicationCategorySelector() {
+    new eXo.social.webui.UIApplicationCategorySelector($params);
+  }
+</script>
 
 <div id="$uicomponent.id" class="ItemListContainer">
   <div class="ItemListTitle ClearFix">
@@ -40,7 +64,7 @@
         itemClass = "SelectedItem Item";
       }
   %>
-    <div title="$applicationCategory.displayName" onclick="$clickAction" class="$itemClass" onmouseover="eXo.webui.UIItemSelector.onOver(this,true);" onmouseout="eXo.webui.UIItemSelector.onOver(this, false);">
+    <div id="$applicationCategory.displayName" title="$applicationCategory.displayName" onclick="$clickAction" class="$itemClass">
 	  <div class="ItemTitle" id="$applicationCategory.name">
 		<div class="CenterItemTitle">
 		  $applicationCategory.displayName
Index: webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIApplicationCategorySelector.js
===================================================================
--- webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIApplicationCategorySelector.js	(revision 0)
+++ webapp/resources/src/main/webapp/javascript/eXo/social/webui/UIApplicationCategorySelector.js	(revision 0)
@@ -0,0 +1,71 @@
+(function() {
+	var window_ = this,
+	Util = eXo.social.Util,
+	UIItemSelector = eXo.webui.UIItemSelector,
+	applicationCategoryIds = null,
+	applicationCategoryEls = null,
+	allApplicationCategorySize = 0;
+	
+	/**
+	 * UIApplicationCategorySelector constructor.
+	 */
+	function UIApplicationCategorySelector(params) {
+	  UIApplicationCategorySelector.configure(params);
+	  UIApplicationCategorySelector.init();
+	};
+	
+	/**
+	 * Configure parameters.
+	 */
+	UIApplicationCategorySelector.configure = function(params) {
+		applicationCategoryIds = UIApplicationCategorySelector.toArray(params.applicationCategoryIds) || null;
+		allApplicationCategorySize = parseInt(params.allApplicationCategorySize);
+	}
+	
+	/**
+	 * Initialize UIApplicationCategorySelector.
+	 */
+	UIApplicationCategorySelector.init = function() {
+		applicationCategoryEls = [];
+		if(applicationCategoryIds != null) {
+			for (var i = 0; i < allApplicationCategorySize; i++) {
+				applicationCategoryEls[i] = Util.getElementById(applicationCategoryIds[i]);
+			}
+		}
+		
+		if (applicationCategoryEls != null) {
+      for (var i = 0; i < allApplicationCategorySize; i++) {
+        (function(i) {
+          //when this element is displayed
+          if (applicationCategoryEls[i]) {
+            Util.addEventListener(applicationCategoryEls[i], ['mouseover', 'focus'], function(evt) {
+            	UIItemSelector.onOver(this,true);
+            }, false);
+
+            Util.addEventListener(applicationCategoryEls[i], ['mouseout', 'blur'], function(evt) {
+            	UIItemSelector.onOver(this,false);
+            }, false);
+          }
+        })(i);
+      }
+    }
+	}
+	
+	/**
+	 * Convert string to array.
+	 * @param {string} str String to convert.
+	 */
+	UIApplicationCategorySelector.toArray = function(str) {
+		var returnArray = str.split(",");
+		for(var i = 0; i < returnArray.length; i++) {
+			returnArray[i] = returnArray[i].replace(/[\[\]\s]/g, "");
+		}
+		return returnArray;
+	}
+	
+  //expose
+  window_.eXo = window_.eXo || {};
+  window_.eXo.social = window_.eXo.social || {};
+  window_.eXo.social.webui = window_.eXo.social.webui || {};
+  window_.eXo.social.webui.UIApplicationCategorySelector = UIApplicationCategorySelector;
+})();
