Index: component/common/src/main/java/org/exoplatform/social/common/ResourceBundleUtil.java
===================================================================
--- component/common/src/main/java/org/exoplatform/social/common/ResourceBundleUtil.java	(revision 80206)
+++ component/common/src/main/java/org/exoplatform/social/common/ResourceBundleUtil.java	(working copy)
@@ -28,8 +28,15 @@
  */
 public class ResourceBundleUtil {
 
-  private static final MessageFormat messageFormat = new MessageFormat("");
+  private static final ThreadLocal<MessageFormat> messageFormatRef = new ThreadLocal<MessageFormat>() {
+    @Override
+    protected MessageFormat initialValue() {
+      return new MessageFormat("");
+    }
+  };
 
+  private static final String[] EMPTY_MESSAGE_ARGUMENTS = new String[]{};
+
   /**
    * Replace convention arguments of pattern {index} with messageArguments[index].
    * @param message
@@ -38,8 +45,9 @@
    */
   public static String replaceArguments(String message, String[] messageArguments) {
     if (messageArguments == null) {
-      messageArguments = new String[]{};
+      messageArguments = EMPTY_MESSAGE_ARGUMENTS;
     }
+    final MessageFormat messageFormat = messageFormatRef.get();
     messageFormat.applyPattern(message);
     return messageFormat.format(messageArguments);
   }
